File: RIRImageResizing/Classes/Category/UIImage+RIRImageResizing.m
//
//  UIImage+RIRImageResizing.m
//  Pods
//
//  Created by Benjamin Maer on 10/7/16.
//
//

#import "UIImage+RIRImageResizing.h"
#import "RIRResizeImageOperation.h"

#import <ResplendentUtilities/RUConditionalReturn.h>





@implementation UIImage (RIRImageResizing)

#pragma mark - scale
-(nullable UIImage*)rir_scaledImage_with_resizeOperationParameters:(nonnull RIRResizeImageOperationParameters*)parameters
{
    kRUConditionalReturn_ReturnValueNil(parameters == nil, YES);
    
    RIRResizeImageOperation* const image_scale_operation = [[RIRResizeImageOperation alloc] init_with_resizeParameters:parameters image:self];
    
    kRUConditionalReturn_ReturnValueNil(image_scale_operation == nil, YES);
    
    return [image_scale_operation resizedImage];
}


@end

File: RIRImageResizing/Classes/Category-Extensions/UIImage+RIRImageResizing.swift
//
//  UIImage+RIRImageResizing.swift
//  Pods-RIRImageResizing_Example
//
//  Created by Benjamin Maer on 4/15/19.
//

import Foundation

extension UIImage {
    func rirScaledImageOperation(with parameters: RIRResizeImageOperationParameters) -> RIRResizeImageOperation {
        return RIRResizeImageOperation(parameters: parameters, image: self)
    }
    
    func rirScaledImage(with parameters: RIRResizeImageOperationParameters) throws -> UIImage {
        return try rirScaledImageOperation(with: parameters).resizedImage()
    }
}

@objc public extension UIImage {
    @objc(rir_scaledImageOperation_with_parameters:) func rirScaledImageOperationObjCLegacy(with parameters: RIRResizeImageOperationParametersObjCLegacy) -> RIRResizeImageOperationObjCLegacy {
        return RIRResizeImageOperationObjCLegacy(parameters: parameters, image: self)
    }
    
    @objc(rir_scaledImage_with_parameters:) func rirScaledImageObjCLegacy(with parameters: RIRResizeImageOperationParametersObjCLegacy) -> UIImage? {
        return rirScaledImageOperationObjCLegacy(with: parameters).resizedImage
    }
}
